Source/ACE.Database/CharacterDatabase.cs

```diff
@@ -277,8 +277,28 @@ public bool IsNameAvailable(string name)
 
                 // Loads the positions into the player object and resets the landlock id
                 LoadCharacterPositions(c);
-                c.Location = c.Positions[PositionType.Location];
-                c.Location.LandblockId = new LandblockId(c.Location.Cell);
+
+                // Load the best location from the database
+                // This may be needed if a position is removed from the database, after a character has been created.
+                // But should now be totally un-neccssary.
+                var invalidPosition = CharacterPositionExtensions.InvalidPosition(c.Id, PositionType.Undef);
+
+                if (c.Positions.ContainsKey(PositionType.Location))
+                    c.Location = c.Positions[PositionType.Location];
+                else if (c.Positions.ContainsKey(PositionType.LastPortal))
+                    c.Location = c.Positions[PositionType.LastPortal];
+                else if (c.Positions.ContainsKey(PositionType.Sanctuary))
+                    c.Location = c.Positions[PositionType.Sanctuary];
+
+                // Check to see if the database position is valid

```

Comment:
Do X/Y/Z really matter?  Just seems like unnecessary overhead.  Will also likely cause problems in dungeons that go outside the bounds of "normal" X/Y/Z values of the open world.